msu/systems/mod_settings/mod_settings_mod_addon.nut

```diff
@@ -39,4 +39,29 @@
 	{
 		return ::MSU.System.ModSettings.getPanel(this.Mod.getID()).hasSetting(_settingID);
 	}
+
+	function requireSettingValue( _setting, _value )
+	{
+		if (_setting.getValue() != _value)
+		{
+			if (_setting.isLocked())
+			{
+				::MSU.QueueErrors.add("Mod " + this.getMod().getID() + " (" + this.getMod().getName() + ") requires setting \'" + _setting.getID() + "\' of mod \'" + _setting.getMod().getID() + " (" + _setting.getMod().getName() + ")\' to have the value \'" + _value + "\' but it is locked to be \'" + _setting.getValue() + "\'. Lock reason: " + _setting.getLockReason() + ".");
+				return false;
```

Comment:
Please use string formatting this is hideous. You can use getValue().tostring() if that is the issue.

Reply:
Pretty sure `getValue().tostring()` throws an exception if `getValue()` returns a C++ instance e.g. Tile. We've run into this before in our `MSU.Exceptions` situation (see commit 4cd9b6ce520ad637850399b1e351cecfd8650151). So, I'm generally averse to using `format` anywhere where there might be anything whose value type is not certain.